platforms/cuda/src/CudaTorchKernels.cpp

```diff
@@ -84,20 +93,21 @@ double CudaCalcTorchForceKernel::execute(ContextImpl& context, bool includeForce
     if (includeForces) {
         energyTensor.backward();
         torch::Tensor forceTensor = posTensor.grad();
-        cu.setAsCurrent();
-        void* data;
         if (cu.getUseDoublePrecision()) {
             if (!(forceTensor.dtype() == torch::kFloat64))
                 forceTensor = forceTensor.to(torch::kFloat64);
-            data = forceTensor.data_ptr<double>();
+            double* data = forceTensor.data_ptr<double>();
+            networkForces.upload(data);
```

Comment:
@peastman 
I don't get how this works. `forceTensor` and `networkForces` are allocated on a GPU. `CudaArray::upload` calls `cuMemcpyHtoD`, but it should be a device-device copy.